ethers-core/src/utils/mod.rs

```diff
@@ -441,6 +447,34 @@ pub fn eip1559_default_estimator(base_fee_per_gas: U256, rewards: Vec<Vec<U256>>
     (max_fee_per_gas, max_priority_fee_per_gas)
 }
 
+/// Deserializes the input into a U256, accepting both 0x-prefixed hex and decimal strings with
+/// arbitrary precision, defined by serde_json's [`Number`](serde_json::Number).
+pub fn from_int_or_hex<'de, D>(deserializer: D) -> Result<U256, D::Error>
```

Comment:
should we have this in other desers too? no need to do now

Reply:
some desers already use this, but it would be worth checking other types of rpc responses that might have decimal values beyond `u64::MAX`